itests/standalone/basic/src/test/java/org/wildfly/camel/test/kafka/KafkaConsumerIntegrationTest.java

```diff
@@ -138,7 +138,7 @@ public void configure() throws Exception {
         props.put(ProducerConfig.ACKS_CONFIG, "1");
         ClassLoader tccl = Thread.currentThread().getContextClassLoader();
         try {
-            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+            Thread.currentThread().setContextClassLoader(org.apache.kafka.clients.producer.KafkaProducer.class.getClassLoader());
```

Comment:
Just wondering if we even need to set the TCCL? This may have been an old hack when the Kafka class loading 
 internals were not so good.

Seems odd we need it just to create an instance of a Kafka producer.

Reply:
I am quite sure we need to set the TCCL as I did, because before the `setContextClassLoader()` call here, the TCCL is already set to the loader of the test deployment, which clearly would make it impossible for kafka config deserialization to be able to see its own internal classes.

Reply:
Seems Camel internally does the same TCCL switch when it creates the KafkaProducer, so I guess it's ok.

IMO it'd be better if Kafka used the ClassLoader of KafkaProducer before falling back on the TCCL. Then there would be no need to switch the TCCL in most cases.